cli/flox/doc/flox-upgrade.md

```diff
@@ -14,31 +14,37 @@ flox-upgrade - upgrade packages in an environment
 ```
 flox [<general-options>] upgrade
      [-d=<path> | -r=<owner>/<name>]
+     [--dry-run]
      [<package or pkg-group>]...
 ```
 
 # DESCRIPTION
 
-Upgrade packages in an environment to versions present in the catalog.
+Apply available upgrades to packages in the environment
 
-When no arguments are specified, all packages in the environment are upgraded.
+When no arguments are specified,
+all packages in the environment with build changes are upgraded.
+Build changes may include version changes, dependency changes,
+or package configuration changes.
 
-Packages to upgrade can be specified by either pkg-group name, or by ID.
-If the specified argument is both a pkg-group name and an install ID,
-both the package with the install ID
-and packages belonging to the pkg-group are upgraded.
+Packages to upgrade can be specified by group name.
+Packages without a specified pkg-group in the manifest
+are placed in a group named 'toplevel'.
+The packages in that group can be upgraded without updating any other groups
+by passing 'toplevel' as the group name.
 
-Packages without a specified pkg-group in the manifest are placed in a
-pkg-group named 'toplevel'.
-The packages in that pkg-group can be upgraded without updating any other
-pkg-groups by passing 'toplevel' as the pkg-group name.
+A single package can only be specified to upgrade by ID
+if it is not in a group with any other packages.
```

Comment:
suggestion nonblocking:
```
A single package can be upgraded by ID if it is not in a group with any other packages.

If a package is in a group with other packages, it must be upgraded by specifying the group name.
Packages without a specified pkg-group in the manifest
are placed in a group named 'toplevel'.
The packages in that group can be upgraded without updating any other groups
by passing 'toplevel' as the group name.
```

Reply:
I'm kinda indifferent about this, maybe we can fllow up on this, resolving as non-blocking for now

cc @jennymahmoudi